core/web/presenters/pipeline_run.go

```diff
@@ -0,0 +1,127 @@
+package presenters
+
+import (
+	"encoding/json"
+	"math/big"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/logger"
+
+	"github.com/shopspring/decimal"
+	"github.com/smartcontractkit/chainlink/core/services/pipeline"
+)
+
+// Corresponds with models.d.ts PipelineRun
+type PipelineRunResource struct {
+	JAID
+	Outputs      []*string                 `json:"outputs"`
+	Errors       []*string                 `json:"errors"`
+	Inputs       pipeline.JSONSerializable `json:"inputs"`
+	TaskRuns     []PipelineTaskRunResource `json:"taskRuns"`
+	CreatedAt    time.Time                 `json:"createdAt"`
+	FinishedAt   time.Time                 `json:"finishedAt"`
+	PipelineSpec PipelineSpec              `json:"pipelineSpec"`
+}
+
+// GetName implements the api2go EntityNamer interface
+func (r PipelineRunResource) GetName() string {
+	return "pipelineRun"
+}
+
+func NewPipelineRunResource(pr pipeline.Run) PipelineRunResource {
+	var trs []PipelineTaskRunResource
+	for i := range pr.PipelineTaskRuns {
+		trs = append(trs, NewPipelineTaskRunResource(pr.PipelineTaskRuns[i]))
+	}
+	// The UI expects all outputs to be strings.
+	var outputs []*string
+	if !pr.Outputs.Null {
+		outs := pr.Outputs.Val.([]interface{})
+		for _, out := range outs {
+			switch v := out.(type) {
+			case string:
+				s := v
+				outputs = append(outputs, &s)
+			case map[string]interface{}:
```

Comment:
This is the main reason for the presenter - to be able to take a map task output and convert to a string. @jkongie you may have opinions on this